diff --git a/options/options.c b/options/options.c
index 716c3e6..71a8333 100644
--- a/options/options.c
+++ b/options/options.c
@@ -245,6 +245,8 @@ static const m_option_t mp_vo_opt_list[] = {
 #if HAVE_EGL_ANDROID
     {"android-surface-size", OPT_SIZE_BOX(android_surface_size)},
 #endif
+    {"external-surface-size", OPT_SIZE_BOX(android_surface_size),
+        .force_update = true},
 #if HAVE_D3D11
     {"d3d11-composition-size", OPT_SIZE_BOX(d3d11_composition_size)},
 #endif
